prometheus/src/main/scala/zio/metrics/prometheus2/package.scala

```diff
@@ -24,36 +24,46 @@ package object prometheus2 {
       }
     }
 
-    private class ServiceImpl(registry: jp.CollectorRegistry, lock: Semaphore) extends Service {
+    private final class ServiceImpl(registry: jp.CollectorRegistry, lock: Semaphore) extends Service {
       def collectorRegistry: UIO[jp.CollectorRegistry] = ZIO.succeed(registry)
-      def updateRegistry[A](f: jp.CollectorRegistry => zio.Task[A]): zio.Task[A] = lock.withPermit {
+
+      def updateRegistry[A](f: jp.CollectorRegistry => Task[A]): Task[A] = lock.withPermit {
         f(registry)
       }
+
       def collect: zio.UIO[ju.Enumeration[jp.Collector.MetricFamilySamples]] =
         ZIO.effectTotal(registry.metricFamilySamples())
     }
+    private object ServiceImpl {
+      def makeWith(registry: jp.CollectorRegistry): UIO[ServiceImpl] =
+        Semaphore
+          .make(permits = 1)
+          .map(new ServiceImpl(registry, _))
+    }
 
-    def live: ULayer[Registry] = ZLayer.fromEffect(
-      Semaphore.make(1).map(new ServiceImpl(new jp.CollectorRegistry(), _))
-    )
+    def live: ULayer[Registry] = ServiceImpl.makeWith(new jp.CollectorRegistry()).toLayer
 
-    def default: ULayer[Registry] = ZLayer.fromEffect(
-      Semaphore.make(1).map(new ServiceImpl(jp.CollectorRegistry.defaultRegistry, _))
-    )
+    def default: ULayer[Registry] = ServiceImpl.makeWith(jp.CollectorRegistry.defaultRegistry).toLayer
 
-    def defaultMetrics: ZLayer[Registry, Throwable, Registry] = ZLayer.fromServiceM { registry =>
-      registry.updateRegistry(r => ZIO.effect(jp.hotspot.DefaultExports.register(r))).as(registry)
+    def provided: URLayer[jp.CollectorRegistry, Registry] = ZLayer.fromFunctionM(ServiceImpl.makeWith)
```

Comment:
I call this `provided` because I feel it's more appropriate than `explicit` from the original API.

Also, I feel the R type parameter for the ZLayer can be simply `jp.CollectorRegistry`, instead of `Has[Option[jp.CollectorRegistry]]` like in the original API. I feel, if one opts to use this constructor and wants to pass in a `jp.CollectorRegistry` explicitly, then wrapping it in a `Some` seems redundant. Similarly, perhaps I miss the original intent, I don't see the need for wrapping  `jp.CollectorRegistry` in a `Has` - feel just  `jp.CollectorRegistry` should suffice and probably be more ergonomic.

Let me know your thoughts. @toxicafunk